### ISSUE TITLE
[Bug] Convert wind speed from OWM from 10m to 2m

### ISSUE THREAD

**Issue by mathijsmeulendijks** [jeroenterheerdt/HAsmartirrigation#57]
**Describe the bug**
PyETO requires a wind velocity at 2m height above ground (https://pyeto.readthedocs.io/en/latest/api.html#evapotranspiration). The OWM API returns wind speeds at 10m height, resulting in an overestimated wind speed due to the atmospheric boundary layer, yielding an overestimated evapotranspiration value in the process. 
I found HASmartIrrigation to greatly overestimate the ET0 values, after debugging I noticed the wind speeds were extremely high compared to experienced wind. I checked the values from the API with KNMI measurements for the Netherlands and other measurement sites to confirm. 

**Expected behavior**

- Convert the OWM API wind speed to the right value using PyETO's wind_speed_2m (https://pyeto.readthedocs.io/en/latest/api.html#wind-speed). 
- Add documentation for sensor users to convert their wind measurements to the right height.

**Comment by jeroenterheerdt** [comment]
Hi @mathijsmeulendijks - thanks for reporting this. Can you please point me to where the OWM API docs say that the windspeed is at 10m height? I am unable to track that down.

**Comment by mathijsmeulendijks** [comment]
Hi Jeroen, the OWM API docs don't include detailed meteorological information, it's oriented more towards developers than meteorologists. Based on my experience with raw weather forecast data I assumed the wind is forecasted at 10m, as this is an industry standard. After checking this assumption with weather data from other well documented sources, I reached out to OWM support. They confirmed that they provide wind speeds at 10m height:
![image](https://user-images.githubusercontent.com/1238016/117703403-e432a800-b1c9-11eb-8303-07e998e6f7b9.png)
I've asked them to include this information in their docs.

**Comment by jeroenterheerdt** [comment]
great, thanks for clarifying and doing the hard work of reaching out to them!

**Comment by jeroenterheerdt** [comment]
Fixed in this commit: https://github.com/jeroenterheerdt/HAsmartirrigation/commit/fe3bb45c1cd7e96f81b41f0662c721af3d38b9a9

**Comment by jeroenterheerdt** [comment]
@mathijsmeulendijks just wanted to check in to see if this worked for you and improved the situation :)

**Comment by mathijsmeulendijks** [comment]
Hey Jeroen, 
I think this definitely helps, wind speeds change quite a bit because of the atmospheric boundary layer. After updating to the latest version I immediately saw a decrease in evapotranspiration for my setup. 

I started to look at the input values for the ETo calculation because I felt like the system overestimated my evapotranspiration. I'm using it to determine irrigation times for approx 2 hectares, overestimates quickly leads to very long watering times and thus higher energy consumption. In the long run I should probably switch to sensor based calculations ;).

In the meantime I'll stick to data based, even though evapotranspiration stays high. For a day like today near Eindhoven the system forecasts ETo=4.23mm, which seems high compared to reference values for mid summer (3.45mm) with higher temperatures and sun. I'm looking for a source to verify the calculated values, but haven't found a decent one yet. 

A further improvement might be to use average hourly wind speeds: looking at the raw OWM data, I see that the predicted wind daily is equal to the maximum wind speed in the hourly forecast. So a single windy hour increases the ETo for the whole day significantly. An alternative might be to use the average hourly wind speeds, but I don't know if that works for Penman Monteith.
For today, the maximum values for wind speed (which is also used for the daily forecast) is 5.64 ms^-1. The average value of hourly wind speeds is 2.94 ms^-1. Using [ETo calculator](http://www.fao.org/land-water/databases-and-software/eto-calculator/en/) I calculated the expected evapotranspiration, resulting in a difference of 0.5mm.

Taking this even further, another improvement would be to determine the daily adjusted values based on observations instead of forecasts. OWM's onecall API also provides [historical data](https://openweathermap.org/api/one-call-api#history) for the lasts 5 days. We could use these values to calculate the hourly ETo, sum them and use the result for further calculations. This way, the daily adjusted runtime is based on actual measurements (which seem to match the KNMI weather station measurements for my region). For the hourly adjusted values we can use observations for past timestamps and hourly forecasts for the future. This gives us increasing accuracy as the day progresses, but retains the property of predicting the whole day. Sidenote: add properties to the sensors to debug this, so show observed ETo from midnight till now and expected ETo based on forecasts separately.
